packages/db/index.js

```diff
@@ -733,4 +733,49 @@ export class DBClient {
         throw new Error('unknown metric requested')
     }
   }
+
+  /**
+   * Get the raw IPNS record for a given key.
+   *
+   * @param {string} key
+   */
+  async resolveNameRecord (key) {
+    /** @type {{ error: Error, data: Array<import('../db-client-types').NameItem> }} */
+    const { data, error } = await this._client
+      .from('name')
+      .select('record')
+      .match({ key })
+
+    if (error) {
+      throw new DBError(error)
+    }
+
+    return data.length ? data[0].record : undefined
```

Comment:
I believe it throws if not found (and I couldn't be bothered to figure out the particular error code it uses) and I want to to return `undefined` if not found.

Reply:
yeah, it will return an error. If this is not what we want, we should rethink some other methods like `getKey`, `getUpload`

Reply:
In general I'm anti throw when things are not found at the data storage layer. In JS it's idiomatic for things to return `undefined` like with `Map`, `Set`,`Array` etc. also with regular `Object`s.

I also wanted to return a better error message that what we'd get from the DB.

Also, we should do this for the other methods because I just saw this in sentry:

<img width="860" alt="Screenshot 2021-11-23 at 11 18 19" src="https://user-images.githubusercontent.com/152863/143015360-bb3a2187-5512-4b6a-86a1-19dea7d13365.png">

Reply:
Yes, that's the `getKey` that I was mentioning. Can you open an issue there for us to change it?

I also prefer to return undefined here

Reply:
Hey, I agree while working on the pinning APIs I've encountered the same problem.
I think in general we should look into abstracting the logic to interpret different DB errors into smaller and more defined ones (ie not found, foreign key violations, connection error etc). Because as you mentioned at the moment everything is caught and on the API level we would send different errors based on the actual returns.